File: python/isogram/isogram.py
alphabet = 'abcdefghijklmnopqrstuvwxyz'

def is_isogram(string:str) -> bool:
    used_letters = []
    for letter in string.lower():
        if letter not in alphabet:
            continue
        if letter in used_letters:
            return False
        used_letters.append(letter)
    return True
    

File: go/isogram/isogram.go
package isogram

import "unicode"

// IsIsogram checks if a word is an isogram (a word with no repeating letters).
// It ignores letter case, hyphens, and spaces.
func IsIsogram(word string) bool {
	letters := make(map[rune]bool)
	for _, char := range word {
		if char == '-' || char == ' ' { // allow hyphens and spaces
			continue
		}
		char = unicode.ToLower(char) // handle case insensitivity

		if letters[char] {
			return false // found a duplicate letter
		}
		letters[char] = true
	}
	return true
}
